src/System.Text.Primitives/System/Text/Parsing/PrimitiveParser2_bool.cs

```diff
@@ -10,6 +10,30 @@ public static partial class PrimitiveParser
     {
         #region helpers
 
+        private static bool IsTrue(ReadOnlySpan<byte> utf8Bytes)
+        {
+            if (utf8Bytes.Length < 4)
+                return false;
+
+            byte firstChar = utf8Bytes[0];
+            if (firstChar != 't' && firstChar != 'T')
```

Comment:
Or to have no branches at all, you can do: return (span.ReadInt32LittleEndian() & ~0x20202020) != 'T' + 0x100 * 'R' + 0x10000 * 'U' + 0x1000000 * 'E';